src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -345,7 +357,12 @@ public function walkJoinCondition(QOM\SelectorInterface $left, QOM\SelectorInter
         }
 
         if ($condition instanceof QOM\EquiJoinConditionInterface) {
-            return $this->walkEquiJoinCondition($left->getSelectorName(), $right->getSelectorName(), $condition);
+            if (!$left instanceof QOM\SelectorInterface) {
```

Comment:
can we invert these for readability? (if $left instanceof then getSelectorName else getJoinCondition...)